src/test/java/com/faforever/client/util/ReflectionUtilTest.java

```diff
@@ -0,0 +1,54 @@
+package com.faforever.client.util;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class ReflectionUtilTest {
+
+
+  @Rule
+  public ExpectedException exceptionGrabber = ExpectedException.none();
+
+  @Test
+  public void testGetDeclaredFieldPublic() throws NoSuchFieldException {
+    Class<?> publicString = ReflectionUtil.getDeclaredField("publicString", TestClass.class);
+    assertThat((publicString == String.class), is(true));
+
+  }
+
+  @Test
+  @Ignore // this will end in a endless loop.
+  public void testGetDeclaredFieldPrivate() throws NoSuchFieldException {
+    /*
+        The Method will first look if the field is inside the given class.
+        If not it will look inside the Superclass.
+        Then, if it's found nothing, it will sets its loop variable (currently the Superclass of the given one),
+        again to the superclass of the given class. This results into a endless loop because every class has Object as
+        superclass (except Object of course) so it will always find Object.class as Superclass and checks like
```

Comment:
Maybe we should fix this then 🤔

Reply:
I think too that it should be. The implemented Code looks like that should not be happening.